modules/firehose/module.go

```diff
@@ -116,6 +117,46 @@ func (m *firehoseModule) Sync(_ context.Context, spec module.Spec) (*resource.St
 	}, nil
 }
 
+func (*firehoseModule) Log(ctx context.Context, spec module.Spec, filter map[string]string) (<-chan module.LogChunk, error) {
+	r := spec.Resource
+
+	var conf moduleConfig
+	if err := json.Unmarshal(r.Spec.Configs, &conf); err != nil {
+		return nil, errors.ErrInvalid.WithMsgf("invalid config json: %v", err)
+	}
+
+	var kubeOut kubernetes.Output
+	if err := json.Unmarshal(spec.Dependencies[keyKubeDependency].Output, &kubeOut); err != nil {
+		return nil, err
+	}
+
+	if filter == nil {
+		filter = make(map[string]string)
+	}
+	filter["app"] = conf.ReleaseConfigs.Name
+
+	kubeCl := kube.NewClient(kubeOut.Configs)
+	logs, err := kubeCl.StreamLogs(ctx, defaultNamespace, filter)
```

Comment:
it would be better to do the error check here before spawning the goroutine and avoid spawning altogether if there is an error.

Reply:
done